log/logcontroller.go

```diff
@@ -1,308 +0,0 @@
-
-	if componentData == nil {
-		globalData, err := globalConfig.Retreive()
-		if err != nil {
-			log.Error(err)
-			return err
-		}
-		globalloglevel := populateLogLevel(globalData)
-		globalConfig.logLevel = globalloglevel
-
-		componentConfig.logLevel = globalConfig.logLevel
-	}
-
-	return nil
-}
-
-//updateComponentLogLevel retrieve the loglevel for the component config from the kvstore then update the  package loglevel with
-//changed loglevel
-func (c *ComponentLogController) updateComponentLogLevel(componentConfig *ComponentConfig, packageName, loglevel string) {
-	data, err := componentConfig.Retreive()
-	if err != nil {
-		log.Error(err)
-	}
-	logLevel := populateLogLevel(data)
-	componentConfig.logLevel = logLevel
-
-	for componentIndex, componentLevel := range componentConfig.logLevel {
-		if componentLevel.PackageName == packageName {
-			componentConfig.logLevel[componentIndex].Level = loglevel
-		}
-	}
-}
-
-//updateLogConfig check the component type is global or component name then update the component config
-func (c *ComponentLogController) updateLogConfig(packageName, logLevel, component string) {
-	//call updateLogConfigToComponent for global to componentname
-	//if default loglevel for global is set and default for component name is not set
-
-	if packageName == "default" && component == "global" {
-		err := c.updateLogConfigToComponent(c.componentNameConfig, c.globalConfig)
-		if err != nil {
-			log.Error(err)
-		}
-	}
-
-	//call updateComponentLogLevel if any changes happen to default or package loglevel for component config
-	if component == c.ComponentName {
-		c.updateComponentLogLevel(c.componentNameConfig, packageName, logLevel)
-	}
-
-}
-
-func (c *ComponentLogController) loadAndApplyLogConfig() {
-	//load the current configuration for component name using Retreive method
-	//create hash of loaded configuration using GenerateLogConfigHash
-	//if there is previous hash stored, compare the hash to stored hash
-	//if there is any change will call UpdateLogLevels
-
-	//generate hash
-	currentLogHash := GenerateLogConfigHash(c.componentNameConfig.logLevel)
-
-	//will set the activeHash to true and update the logHash
-	if c.logHash != currentLogHash {
-		UpdateLogLevels(c.componentNameConfig.logLevel)
-		c.hashActive = true
-		c.logHash = currentLogHash
-	}
-
-}
-
-//call getDefaultLogLevel to get active default log level
-func getDefaultLogLevel(logLevel []LogLevel) string {
-
-	for _, level := range logLevel {
-		if level.PackageName == "default" {
-			return level.Level
-		}
-	}
-	return ""
-}
-
-//create loglevel to set loglevel for the component
-func createCurrentLogLevel(activeLogLevels, currentLogLevel []LogLevel) []LogLevel {
-
-	level := getDefaultLogLevel(currentLogLevel)
-	for activeIndex, activeLevel := range activeLogLevels {
-		exist := false
-		for currentLogLevelIndex, _ := range currentLogLevel {
-			if activeLogLevels[activeIndex].PackageName == currentLogLevel[currentLogLevelIndex].PackageName {
```

Comment:
This logic is only handling the case when you are adding a new package loglevel. It does not seem to handle the cases when a package has been deleted or its loglevel has changed